### PR TITLE
[ControlGallery] Add more CarouselView samples

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Add more CarouselView samples

### API Changes ###
 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by adrianknight89** [comment]
@jsuarezruiz @samhouts Regarding Gallery samples, how often do you run them to verify they are working as intended (i.e. no regression)? I've seen a lot of samples in the past that were broken. What about manual tests? Are they run before each new release?

**Comment by rmarinho** [comment]
@adrianknight89 we run regulary via the uitests, in major releases (true is not always) we go and try manually most of the gallery. 

We improved our uitests lanes to have (core tests, regression tests, manual tests) Where the manual results can be validaded. So the trick here is mark manual so they can be reviewed after and rely on UITests that the others work fine. Also we are looking in screenshot comparison as yet a another way to make sure we don't regress.

Do you have any examples that worry you?

You can also now check our tests here :

https://dev.azure.com/xamarin/public/_releaseProgress?_a=release-pipeline-progress&releaseId=346

**Comment by adrianknight89** [comment]
@rmarinho Thanks. So, just to clarify, the ui tests marked "manual" are verified before each release, correct? 

I've run into several issues with the gallery. For example, if you run the application and immediately go to CollectionView Core Gallery or CarouselView Core Gallery, then the app will crash because the experimental flag has not been set yet. I saw an issue with the update button as mentioned in #7472. I think at some point `IsFocused` gallery samples did not work properly, but I checked that a long time ago.

If it was up to me, I'd get rid of the gallery samples and add everything as manual UI tests with on-screen instructions. Sometimes, it's difficult to deduce whether a gallery sample is working as intended because there are no instructions. The gallery seems to be a playground.

Also, does your Scrum team have a dedicated QA tester or is the burden left on you guys' shoulders?

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CarouselViewGalleries/CollectionCarouselViewGallery.cs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+			{
+				Items.Add(new CarouselData
+				{
+					Color = Color.FromRgb(_random.Next(0, 255), _random.Next(0, 255), _random.Next(0, 255)),
+					Name = $"NewItemsObservable {n + 1}"
+				});
+			}
+		}
+
+		async Task Thead()
```
Comment (samhouts) [inline Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CarouselViewGalleries/CollectionCarouselViewGallery.cs]: Thead?

Comment (hartez) [reply Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CarouselViewGalleries/CollectionCarouselViewGallery.cs]: Yeah, please change this to `Thread` or something more descriptive.

Comment (jsuarezruiz) [reply Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CarouselViewGalleries/CollectionCarouselViewGallery.cs]: Oops, fixed!.

**Pull request merged**